cmd/datasource-syncer/README.md

```diff
@@ -0,0 +1,63 @@
+# Datasource Syncer
+
+This CLI tool can act as a cron job remotely syncing the given Grafana Prometheus datasource, so it has e.g.:
+
+* Correct URL
+* Correct HTTP method
+* Fresh Oauth2 access token (valid for 1h)
+
+This allows you to directly query Google Cloud Monitoring with Grafana.
+
+[Google access tokens have a lifetime of 1 hour.](https://cloud.google.com/docs/authentication/token-types#at-lifetime) This script should be run hourly to ensure you have an uninterrupted connection between Grafana and Google Cloud Monitoring.
+
+### Run
+
+1. [Make sure you have setup your Grafana datasource setup.](https://cloud.google.com/stackdriver/docs/managed-prometheus/query#grafana-datasource) Extract datasource IDs you wish to use from URL.
+
+2. [Setup a Grafana service account](https://grafana.com/docs/grafana/latest/administration/service-accounts/#create-a-service-account-in-grafana) and generate a token.
+
+3. Figure out the URL of your Grafana instance. e.g. "https://your.grafana.net"
+
+Running the following command should update your datasources with a new token:
+
+```bash
+GRAFANA_TOKEN=YOUR_GRAFANA_SERVICE_ACCOUNT_TOKEN
+GRAFANA_ENDPOINT=YOUR_GRAFANA_INSTANCE_URL
+DATASOURCE_UIDS=YOUR_DATASOURCE_UIDs
+```
+```bash
+go run main.go \
+  --grafana-api-token=$GRAFANA_TOKEN \
+  --grafana-api-endpoint=$GRAFANA_ENDPOINT \
+  # Can be a comma separated list of datasource IDs
+  --datasource-uids=$DATASOURCE_UIDS
+```
+### Kubernetes Cron Job
+
+You can run a cron job that runs every 1 hour to refresh your datasources with a new token using datasource-syncer.yaml:
+
+```bash
+cat datasource-syncer.yaml \
+| sed 's|$GRAFANA_TOKEN|'"$GRAFANA_TOKEN"'|; s|$GRAFANA_ENDPOINT|'"$GRAFANA_ENDPOINT"'|; s|$DATASOURCE_UIDS|'"$DATASOURCE_UIDS"'|' \
+| kubectl apply -f -
+```
+
+### Custom Google Cloud Monitoring Service Account and Monitoring Endpoint
+
+To use with a custom Google Cloud Monitoring service account and ensure your datasources GCM target is setup correctly, you can use the following command:
+
+```bash
+PROJECT_ID=PROJECT_ID_TO_QUERY_GCM_WITH
+GCM_TARGET=https://monitoring.googleapis.com/v1/projects/$PROJECT_ID/location/global/prometheus/
+CREDENTIALS=SERVICE_ACCOUNT_WITH_GOOGLE_CLOUD_MONITORING_READ_ACCESS
+```
+
+```bash
+go run main.go \
+  --prometheus-endpoint=$GCM_TARGET \
+   # Create service account instructions: https://cloud.google.com/iam/docs/service-accounts-create and create a key.
+  --json-file-path=YOUR_GOOGLE_CLOUD_SERVICE_ACCOUNT_KEY_FILE_PATH \
```

Comment:
should note that --json-file-path as optional, saying that if not specified it will use the default compute engine service account of the local proejct.